components/ui/AddressInput.tsx

```diff
@@ -0,0 +1,199 @@
+    <components.IndicatorsContainer {...rest} className="pr-4">
+      {children}
+    </components.IndicatorsContainer>
+  );
+};
+
+const DropdownIndicator = (_: DropdownIndicatorProps<AddressOption, false>) => {
+  return <></>;
+};
+
+const ClearIndicator = (props: ClearIndicatorProps<AddressOption, false>) => {
+  return (
+    <components.ClearIndicator
+      {...omit(props, "children")}
+      className="cursor-pointer"
+    >
+      <X />
+    </components.ClearIndicator>
+  );
+};
+
+const MenuList = (props: MenuListProps<AddressOption, false>) => {
+  return (
+    <components.MenuList
+      {...props}
+      className="!w-full !absolute bg-white mt-1 rounded-md"
+    >
+      {props.children}
+    </components.MenuList>
+  );
+};
+
+const SelectContainer = (props: ContainerProps<AddressOption, false>) => {
+  return (
+    <components.SelectContainer {...omit(props, "children")}>
+      {props.children}
+    </components.SelectContainer>
+  );
+};
+
+const ValueContainer = (props: ValueContainerProps<AddressOption, false>) => {
+  return (
+    <components.ValueContainer
+      {...omit(props, "children")}
+      className="!flex items-center h-full"
+    >
+      {props.children}
+    </components.ValueContainer>
+  );
+};
+
+const SingleValue = ({
+  children,
+  ...rest
+}: SingleValueProps<AddressOption, false>) => {
+  const address = rest.data.value;
+  const isValidAddress = isValidPolkadotAddress(address);
+  if (
+    typeof children === "string" &&
+    !children.includes(".") &&
+    isValidAddress
+  ) {
+    children = shortenAddress(children, 13, 13);
+  }
+  return (
+    <components.SingleValue {...rest} className="flex items-center font-medium">
+      {isValidAddress && (
+        <div className="w-9 h-9 mr-3 bg-transparent rounded-full center">
+          <Avatar address={address} size={36} />
+        </div>
+      )}
+      {children}
+    </components.SingleValue>
+  );
+};
+
+const Option = ({ children, ...rest }: OptionProps<AddressOption, false>) => {
+  const { value: address, label } = rest.data;
+  return (
+    <components.Option
+      {...rest}
+      className="!flex items-center w-full h-14 px-4 !cursor-pointer bg-anti-flash-white rounded-md mb-2 last:mb-0"
+    >
+      {isValidPolkadotAddress(address) && (
+        <div className="w-9 h-9 mr-3 bg-transparent rounded-full center">
+          <Avatar address={address} size={36} />
+        </div>
+      )}
+      <div>{label}</div>
+    </components.Option>
+  );
+};
+
+const Input = (props: InputProps<AddressOption, false>) => {
+  return <components.Input {...props} className="absolute w-full left-0" />;
+};
+
+export type AddressSelectProps = {
+  onChange: (option: AddressOption | null) => void;
+  value?: AddressOption | null;
+  error?: string;
+};
+
+const AddressInput: React.FC<AddressSelectProps> = ({
+  onChange,
+  error,
+  value = null,
+}) => {
+  const wallet = useWallet();
+
+  const options = useMemo<AddressOption[]>(() => {
```

Comment:
I would like to maybe reuse this in the proxy selection but with a different set of addresses, so could you refactor the AddressInput to take a prop `options: AddressOption[]` so it can be used with any set of addresses not just the wallet addresses?

Reply:
we talked about this yesterday and we agreed that I'll integrate this component in the settings modal after you do the proxy stuff you made with Account select we use for the wallet.